### ISSUE TITLE
Registration of the workspace is unskippable in version 6.5

### ISSUE THREAD

**Issue by provelo-ict** [RocketChat/Rocket.Chat#31163]
Hello,

I've had an instance of Rocket.Chat running for three years now, and I've never had the need (let alone the desire) to register it as I don't use push notification nor store addons or anything that are limited to subscribed users. 

After upgrading to version 6.5, the admin account get the modal below, without the ability to skip this step or the continue the wizard without registering.

This issue is similar to #31157 but I want to focus on the fact that there should be a way to skip the registration.
This would also be needed for instances in test environments.

![image](https://github.com/RocketChat/Rocket.Chat/assets/56344035/3a2d56c6-7f43-4987-b6e1-cf278601ac02)

### Server Setup Information:

- Version of Rocket.Chat Server: 6.5
- Operating System:  Ubuntu 20.04
- Deployment Method: snap
- Number of Running Instances: 1

**Comment by Gummikavalier** [comment]
Indeed there are circumstances where you simply cannot connect to cloud.rocket.chat for registering. Not even once.

**Comment by 72tlukas72** [comment]
Just look at 2.5k issues. That faith is already gone...

Same problem in airgap with docker.

**Comment by LeeThompson** [comment]
This is extremely lame of RocketChat.

**Comment by maswan** [comment]
I don't appreciate suddenly being cut off from communicating with my team, especially when we have to discuss if the terms and conditions and privacy policy for the cloud account are acceptable to our organization.

**Comment by florisvangeel** [comment]
If you set the environment variables [properly](https://youtu.be/JIr9aMTiCWo?list=PLee3gqXJQrFXRW6hZWGUM2pMKQejChDeI&t=961) for the registration fields needed in this wizard then the option environement flag to `OVERWRITE_SETTING_Show_Setup_Wizard=completed` should still be [available](https://github.com/search?q=repo%3ARocketChat%2FRocket.Chat+Show_Setup_Wizard&type=code&p=1) in 6.5.0

**Comment by provelo-ict** [comment]
> If you set the environment variables [properly](https://youtu.be/JIr9aMTiCWo?list=PLee3gqXJQrFXRW6hZWGUM2pMKQejChDeI&t=961) for the registration fields needed in this wizard then the option environement flag to `OVERWRITE_SETTING_Show_Setup_Wizard=completed` should still be [available](https://github.com/search?q=repo%3ARocketChat%2FRocket.Chat+Show_Setup_Wizard&type=code&p=1) in 6.5.0


Thanks a lot @florisvangeel  !

So for someone using snap, it means creating a file 
`/var/snap/rocketchat-server/common/override-setup-wizard.env ` (the name of the file itself could be anything as long as it has an .env extension) and setting its content to 
`OVERWRITE_SETTING_Show_Setup_Wizard=completed`

Then, restarting the server by
`systemctl restart snap.rocketchat-server.rocketchat-server.service`

**Comment by florisvangeel** [comment]
Seems this issue can be closed.
Thank you for the continuous great work @Rocket.Chat Team.

**Comment by TLINDEN** [comment]
Doesn't work for me. I have set the environment variable:
```shell
I have no name!@rocketchat-fitsintegration-rocketchat-5478d9df75-dw8hh:/app/bundle$ echo $OVERWRITE_SETTING_Show_Setup_Wizard
completed
```

But when I visit the page after the update, there's still the registration wizard active and wants me to complete the process w/o any "skip" function anywhere.

**Comment by amsnek** [comment]
OVERWRITE_SETTING_Show_Setup_Wizard=completed 
-> this does not work anymore in 6.6.x?

**Comment by provelo-ict** [comment]
> OVERWRITE_SETTING_Show_Setup_Wizard=completed -> this does not work anymore in 6.6.x?

Works for me in 6.6.2, but I have to restart the service after every update to make the Wizard disappear.

**Comment by TLINDEN** [comment]
I added this setting, but the wizard came up anyway.

**Comment by amsnek** [comment]
you need to restart rocketchat
i also have additionally the environment setting:
OVERWRITE_SETTING_Show_Setup_Wizard=completed 

-> but that alone didnt suffice 
I have rolled back from 6.6.x to 6.5.x because a lot of chat histories *freeze* the browser/client when updated from 6.5.3 -> 6.6.3

**Comment by TLINDEN** [comment]
> you need to restart rocketchat

Yeah, I'm deploying the helm chart with ansible to K8S, so it's being restarted anyway.

But I saw, that going from 6.5.x => up the setting works. It only doesn't work going from 6.4.x => 6.5.

> -> but that alone didnt suffice (anymore) I have rolled back from 6.6.x to 6.5.x because a lot of chat histories _freeze_ the browser/client when entering such chats after I updated from 6.5.3 -> 6.6.3

Oh, good to know, is there an issue for this already?

**Comment by amsnek** [comment]
No issue for that, i have seen some commuity reports, but I am still uncertain about the possible causes. Initially i thought it was related to (historic) Jitsi Calls in those rooms/chats -> but I am not sure anymore, still trying to pinpoint. I will try a new update approach from latest 6.5.4 -> 6.60.
What seems a bit odd though, it doesnt do the usual "run migrations"

**Comment by amsnek** [comment]
going from 6.5.4 -> 6.0.0 works

**Comment by florisvangeel** [comment]
true, there is no 'substantial changes' between 6.5.4 and 6.6.0 so no migrations needed.

This combination of client (app) [enforced upgrades](https://github.com/RocketChat/Rocket.Chat.Electron/blob/bdcc4347aadca79c7df8ba04615dcf49940bfbb9/src/servers/supportedVersions/main.ts#L86) (current <12 days to 6.5.3) and unstable/funneled new version 6.6.x that enrols automatically into a subscription of 'Starter Licence' is deadly for user-land and potentially also for community usage.

Until this is fixed i have to stick to 6.5.x and change API reported version to match the client check to keep the app behaving.

**Comment by amsnek** [comment]
@florisvangeel that is an interesting approach, how do change the reported version for the API? adjusting the contents from /api/info?

**Comment by smirnovegorv** [comment]
Horrible behavior. Truely horrible. Setting `OVERWRITE_SETTING_Show_Setup_Wizard: completed` in my docker-compose.yml environment section worked, but thats the last call. Each Rocket Chat update now breaks something, removes features (like read receitps when updating from older versions) and forces cloud registration now. That's really a sign to switch to some other chat.

**Comment by florisvangeel** [comment]
> @florisvangeel that is an interesting approach, how do you change the reported version for the API? adjusting the contents from /api/info?

yes i still need to test this method also in relation to future `working` updates, it would drill down to [getServerInfo.ts](https://github.com/RocketChat/Rocket.Chat/blob/78e95abaa0cd7f8be02d1d827783cd268a7f2fc0/apps/meteor/app/api/server/lib/getServerInfo.ts#L17) or [one level deeper](https://github.com/RocketChat/Rocket.Chat/blob/78e95abaa0cd7f8be02d1d827783cd268a7f2fc0/apps/meteor/app/utils/rocketchat.info)

**Comment by amsnek** [comment]
@florisvangeel k, thanks! I am pondering if I should provide a "customized" /api/info on the reverse proxy for electron clients in the future (minimumClientVersions). Not sure if i would want to adjust the "version" (server) itself

**Comment by florisvangeel** [comment]
> @florisvangeel k, thanks! I am pondering if I should provide a "customized" /api/info on the reverse proxy for electron clients in the future (minimumClientVersions). Not sure if i would want to adjust the "version" (server) itself

me neither, still need to test this route with updates.

**Comment by florisvangeel** [comment]
@amsnek i like your idea to tackle this for the time being in the proxy
serving a single [json](https://open.rocket.chat/api/info) statically on /api/info buys us some time to fix this properly.

**Comment by jittygitty** [comment]
I just found this thread. For those of you who had trouble bypassing registration and agreement to their cloud privacy policy, please make sure that you create the environment variable set to completed for the user the user that rocketchat service is running under (usually 'rocketchat'?). Since setting that environment variable worked for me. See my issue:
https://github.com/RocketChat/Rocket.Chat/issues/32658

Does anyone have confirmation that setting that OVERWRITE_SETTING_Show_Setup_Wizard=completed  actually does OPT-OUT of any data sharing to their cloud and doesn't just make the codebase think that you already "AGREED" and completed their form to agree to their privacy policy and data sharing etc? Would be great if someone can confirm this.

**Comment by Gummikavalier** [comment]
Traditionally at least RocketChat has called to home in many ways:
One of the clearest is the use of go.rocket.chat help links on the home page call home from the client addresses.

Also marketplace, online registration and update checks actively use RocketChat cloud services.

Home page used to have direct links to call for different kind of registration schemes but those have been removed.

At one point there were also several trackers assigned but not anymore; I think the RC devs were was just tracking out their user base size to consider the their sales tactics.

**Comment by amsnek** [comment]
Yes, while I do not know what RC wants to access (probably things like latest version, Marketplace information, etc ?) it does attempt to connect to the internet. For me, the UID under which it runs is not allowed to create outgoing connections.

**Comment by jittygitty** [comment]
Thanks for the info, any of you using portmaster or opensnitch or other application firewall to more easily detect such outgoing traffic? amsnek using iptables -A OUTPUT -m owner --uid-owner rocketchat -j DROP ?

**Comment by amsnek** [comment]
in my case, it is just with iptables with OUTPUT -m owner --uid <rocketchat>, with REJECT and not DROP howerver, else the application will hang quite a while on startup. It can reply to established incoming connections of course from the reverse proxy :)

**Comment by jittygitty** [comment]
@amsnek Great! That sounds like a good setup, thanks for the tip!

**Comment by Gummikavalier** [comment]
Note that if you use message previews, fetching and rendering those happens on the server side. So you should be specific on what connections to cut.
(If you test this out old messages previews keep working as they are cached in the database for a certain time.)

**Comment by amsnek** [comment]
good point. Previews (I assume you mean loading images, videos or apps like giphy etc) are disabled in my setup though, so that is not an issue. Might still be possible to reject/ristrict outgoing traffic with a (whitelist) proxy though.

**Comment by jittygitty** [comment]
I purposefully disable previews etc if there's option to, so I guess this limitation would be a "feature" for me?

**Comment by jethrogillgren** [comment]
Is anyone having issues using the `OVERWRITE_SETTING_Show_Setup_Wizard: completed` variable on new versions of RocketChat?
My old instance (6.7) skipped the registration OK with the ENV var set as above, but new instances are still forcing the Setup-wizard...

**Comment by amsnek** [comment]
just the environment variable did not suffice for me. 
use mongoshell to connect to rocketchat db and adjust it in the database as well
```
use.rocketchat
db.rocketchat_settings.update({"_id":"Show_Setup_Wizard"}, {$set: {"value" : "completed"} });
```

**Comment by sharovmerk** [comment]
I think that it should be added to official doc

**Comment by pysiekytel** [comment]
While [official docs mention](https://docs.rocket.chat/docs/admin-account-creation#:~:text=OVERWRITE_SETTING_Show_Setup_Wizard%3Dcompleted) that setting `OVERWRITE_SETTING_Show_Setup_Wizard=completed` is possible in reality it doesn't work. This should be fixed.

**Comment by etienne-monier** [comment]
+1

**Comment by pedoc** [comment]
not working with 7.9.3

**Comment by Cloud-Kid** [comment]
+1

**Comment by Abhishek2005-ard** [comment]
Hi maintainers,
I would like to work on this issue. I’ll start investigating and share progress here.
Thanks!

**Comment by vincgj** [comment]
Hi there, any update on this ?

We ran into the same issue with 7.13.2

We cannot restore any backup as all are leading to the registration page after a restart :/

We cannot even go to step 2 of the wizard

The `Show_Setup_Wizard` is already marked as "completed" in mongodb

Tried adding env with `ADMIN_*` stuff as suggested earlier with no luck.

Any help or advice appreciated !

---

edit: solved for us 

Cause: issue with existing / old auth token in local browser

Resolution: Clear the cache of your browser to start a new auth from scratch

---

edit2: created https://github.com/RocketChat/Rocket.Chat/issues/38467
